dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/S3Store.java

```diff
@@ -0,0 +1,68 @@
+package com.salesforce.dockerfileimageupdate.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.salesforce.dockerfileimageupdate.utils.DockerfileGitHubUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+
+public class S3Store implements ImageTagStore {
+    private static final Logger log = LoggerFactory.getLogger(S3Store.class);
+    private static final Character S3_FILE_KEY_PATH_DELIMITER = ';';
+    private final AmazonS3 s3;
+    private final String store;
+
+    public S3Store(AmazonS3 s3, String store) {
+        this.s3 = s3;
+        this.store = store;
+    }
+
+    /* The store link should be a bucket name on S3. */
+    public void updateStore(String img, String tag) throws IOException {
+        if (store == null) {
+            log.info("Image tag store cannot be null. Skipping store update...");
+            return;
+        }
+        log.info("Updating store: {} with image: {} tag: {}...", store, img, tag);
+        if (!s3.doesBucketExistV2(store)) {
+            throw new IOException(String.format("The S3 bucket with name %s does not exist. Cannot proceed.", store));
+        }
+        String key = convertImageStringToS3ObjectKey(img);
+        s3.putObject(store, key, tag);
+    }
+
+    public HashMap<String, String> getStoreContent(DockerfileGitHubUtil dockerfileGitHubUtil, String storeName) {
+        HashMap<String, String> imageNameWithTag = new HashMap<>();
+        ListObjectsV2Result result = getS3Objects();
+        List<S3ObjectSummary> objects = result.getObjectSummaries();
+        for (S3ObjectSummary os : objects) {
+            String key = os.getKey();
+            String image = convertS3ObjectKeyToImageString(key);
+            S3Object o = getS3Object(store, key);
```

Comment:
We might get throttled by S3 here, so we might need to inject a short sleep in this loop.